lib/openfeature/sdk/client.rb

```diff
@@ -1,18 +1,83 @@
 # frozen_string_literal: true
 
+require "concurrent"
+require "forwardable"
+
 module OpenFeature
   module SDK
     # TODO: Write documentation
     #
     class Client
+      extend Forwardable
+
       attr_reader :metadata
 
       attr_accessor :hooks
 
-      def initialize(provider, client_options, context)
+      def initialize(provider:, client_options: nil, context: nil)
         @provider = provider
-        @client_options = client_options
+        @metadata = client_options
         @context = context
+        @hooks = Concurrent::Array.new([])
+      end
+
+      def fetch_boolean_value(flag_key:, default_value:, evaluation_context: nil, evaluation_options: nil)
+        evaluate_flag_details(flag_type: :boolean, flag_key: flag_key, default_value: default_value,
+                              evaluation_context: evaluation_context, evaluation_options: evaluation_options)
+      end
```

Comment:
All of these methods seem pretty similar, only varying by the type (ie boolean) and value or string. And I'm not seeing any difference between `value` and `details` methods 🤔 

Maybe could define these in a loop?


```ruby
      RESULT_TYPE = %i(boolean string  number object).freeze
      SUFFIXES = %i(value details)
      TYPES.each do |type|
        SUFFIXES.each do |suffix|
        
          define_method "fetch_#{type}_#{suffix}, ...
```

Reply:
Worked with @technicalpickles to implement this